src/test/components/TrackThread.test.js

```diff
@@ -0,0 +1,173 @@
+
+describe('timeline/TrackThread', function() {
+  function getSamplesProfile() {
+    return getProfileFromTextSamples(`
+      a  d  g  j
+      b  e  h  k
+      c  f  i  l
+    `).profile;
+  }
+
+  function getMarkersProfile() {
+    const profile = getProfileWithMarkers([
+      ['Marker A', 0, { startTime: 0, endTime: 1 }],
+      ['Marker B', 1, { startTime: 1, endTime: 2 }],
+      ['Marker C', 2, { startTime: 2, endTime: 3 }],
+      ['Marker D', 3, { startTime: 3, endTime: 4 }],
+    ]);
+    const [thread] = profile.threads;
+    thread.name = 'GeckoMain';
+    thread.processType = 'default';
+    return profile;
+  }
+
+  function setup(profile: Profile) {
+    const store = storeWithProfile(profile);
+    const { getState, dispatch } = store;
+    const threadIndex = 0;
+    const flushRafCalls = mockRaf();
+    const ctx = mockCanvasContext();
+    jest
+      .spyOn(HTMLCanvasElement.prototype, 'getContext')
+      .mockImplementation(() => ctx);
+
+    jest
+      .spyOn(HTMLElement.prototype, 'getBoundingClientRect')
+      .mockImplementation(() => getBoundingBox(400, 50));
+
+    const view = mount(
+      <Provider store={store}>
+        <TrackThread threadIndex={threadIndex} />
+      </Provider>
+    );
+
+    // WithSize uses requestAnimationFrame
+    flushRafCalls();
+    view.update();
+
+    const stackGraphCanvas = view.find('.timelineStackGraphCanvas').first();
+    const tracingMarkersCanvas = view
+      .find(
+        [
+          '.timelineTrackThreadIntervalMarkerOverviewThreadGeckoMain',
+          '.timelineTracingMarkersCanvas',
+        ].join(' ')
+      )
+      .first();
+
+    return {
+      dispatch,
+      getState,
+      profile,
+      thread: profile.threads[0],
+      store,
+      view,
+      threadIndex,
+      stackGraphCanvas,
+      tracingMarkersCanvas,
+    };
+  }
+
+  it('matches the snapshot', () => {
+    const { view } = setup(getSamplesProfile());
+    expect(view).toMatchSnapshot();
+  });
+
+  it('has the correct selectors into useful parts of the component for the samples profile', function() {
+    const { stackGraphCanvas } = setup(getSamplesProfile());
+    expect(stackGraphCanvas.exists()).toBe(true);
+  });
+
+  it('has the correct selectors into useful parts of the component for the markers profile', function() {
+    const { tracingMarkersCanvas } = setup(getMarkersProfile());
+    expect(tracingMarkersCanvas.exists()).toBe(true);
+  });
+
+  it('can click a stack in the stack graph', function() {
+    const { getState, stackGraphCanvas, thread } = setup(getSamplesProfile());
+
+    // Provide a quick helper for nicely asserting the call node path.
+    const getCallNodePath = () =>
+      selectedThreadSelectors
+        .getSelectedCallNodePath(getState())
+        .map(funcIndex =>
+          thread.stringTable.getString(thread.funcTable.name[funcIndex])
+        );
+
+    stackGraphCanvas.simulate('mouseup', getMouseEvent({ pageX: 50 }));
```

Comment:
the `50` here seems arbitrary, could we have a constant to make it more explicit ? (that would work for the following pageX used in this test )

Reply:
Good point on the magicness of the values, I collected them at the top with a comment.

```
// The stack graph is 400 pixels wide based on the getBoundingBox mock. Each stack is
// 100 pixels wide. Use the value 50 to click in the middle of this stack, and
// incrementing by steps of 100 pixels to get to the next stack.
const GRAPH_WIDTH = 400;
const GRAPH_HEIGHT = 50;
const STACK_1_X_POSITION = 50;
const STACK_2_X_POSITION = 150;
const STACK_3_X_POSITION = 250;
const STACK_4_X_POSITION = 350;
``